drivers/serial/uart_stm32.c

```diff
@@ -799,16 +839,30 @@ static void uart_stm32_dma_rx_flush(const struct device *dev)
 static void uart_stm32_isr(const struct device *dev)
 {
 	struct uart_stm32_data *data = DEV_DATA(dev);
+#if defined(CONFIG_PM) || defined(CONFIG_UART_ASYNC_API)
```

Comment:
below, there is also a `uart_stm32_isr() `for the CONFIG_PM

Reply:
The other `uart_stm32_isr` is specific for !CONFIG_UART_ASYNC_API && ! IRQ_DRIVEN

Reply:
yes, removed in the next commit